src/components/Calculator.js

```diff
@@ -1,36 +1,51 @@
 import React from 'react';
 import './Calculator.css';
+import calculate from '../logic/calculate';
 
 class Calculator extends React.Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.state = {
+      total: 0,
+      next: null,
+      operation: null,
+    };
+    this.clickEvent = this.clickEvent.bind(this);
+  }
+
+  clickEvent(e) {
+    this.setState((state) => calculate(state, e.target.textContent));
   }
 
   render() {
+    const { next, total, operation } = this.state;
     return (
       <main>
-        <div className="output">0</div>
+        <div className="output">
+          {total}
+          {operation}
+          {next}
+        </div>
```

Comment:
[OPTIONAL] When the user press AC, the state of total goes blank, hence the screen does not show any number. I believe the standard behavior, in this case, is to display a `0` that represents the stand-by state of the calculator. It would be great if you could reflect this behavior on your calculator as well as it will bring a more familiar experience to the user.